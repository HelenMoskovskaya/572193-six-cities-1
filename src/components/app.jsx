import React from "react";
import MainPage from './main-page.jsx';
import PropTypes from "prop-types";

const App = (props) => {
  const {cards} = props;
  return <MainPage
    cards = {cards}
  />;
};

App.propTypes = {
  cards: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default App;
