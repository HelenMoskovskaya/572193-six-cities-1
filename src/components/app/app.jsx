import React from 'react';
import MainPage from '../main-page/main-page.jsx';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {ActionCreatorData} from '../../reducer/data/data.js';
import {getActiveCity, getCityList, getActiveOffers} from '../../reducer/data/selectors.js';
import {getAuthorizationStatus, getUserData} from '../../reducer/user/selectors.js';
import {Operation} from '../../reducer/user/user';
import Header from '../header/header.jsx';
import Svg from '../svg/svg.jsx';

import SignIn from '../sign-in/sign-in.jsx';

const App = (props) => {
  const {offers, city, onCityClick, cities, isAuthorizationRequired, loginUser, userData} = props;
  const mainPage = <MainPage offers = {offers} cities = {cities} onCityClick={onCityClick} city={city}/>;
  const signIn = <SignIn loginUser={loginUser}/>;

  return <div className={`page page--gray ${isAuthorizationRequired ? `page--main` : `page--login`}`}>
    <Svg />
    <Header
      isAuthorizationRequired = {isAuthorizationRequired}
      userData = {userData}
    />
    {isAuthorizationRequired ? mainPage : signIn};
  </div>;
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  city: getActiveCity(state),
  offers: getActiveOffers(state),
  cities: getCityList(state),
  isAuthorizationRequired: getAuthorizationStatus(state),
  userData: getUserData(state)
});

const mapDispatchToProps = (dispatch) => ({
  onCityClick: (activeCity) => {
    dispatch(ActionCreatorData.changeCity(activeCity));
  },

  loginUser: (email, password) => {
    dispatch(Operation.authorizeUser(email, password));
  }
});

App.propTypes = {
  offers: PropTypes.arrayOf(PropTypes.shape({
    city: PropTypes.shape({
      name: PropTypes.string.isRequired,
      location: PropTypes.shape({
        latitude: PropTypes.number.isRequired,
        longitude: PropTypes.number.isRequired,
        zoom: PropTypes.number.isRequired
      })
    }),
    previewImage: PropTypes.string.isRequired,
    images: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
    title: PropTypes.string.isRequired,
    isFavorite: PropTypes.bool.isRequired,
    isPremium: PropTypes.bool.isRequired,
    rating: PropTypes.number.isRequired,
    type: PropTypes.string.isRequired,
    bedrooms: PropTypes.number.isRequired,
    maxAdults: PropTypes.number.isRequired,
    price: PropTypes.number.isRequired,
    goods: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
    host: PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      isPro: PropTypes.bool.isRequired,
      avatarUrl: PropTypes.string.isRequired
    }),
    description: PropTypes.string.isRequired,
    location: PropTypes.shape({
      latitude: PropTypes.number.isRequired,
      longitude: PropTypes.number.isRequired,
      zoom: PropTypes.number.isRequired
    }),
    id: PropTypes.number.isRequired
  })).isRequired,

  city: PropTypes.string.isRequired,
  cities: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  onCityClick: PropTypes.func.isRequired,
  isAuthorizationRequired: PropTypes.bool.isRequired,
  loginUser: PropTypes.func.isRequired,
  userData: PropTypes.object.isRequired
};

export {App};
export default connect(mapStateToProps, mapDispatchToProps)(App);
