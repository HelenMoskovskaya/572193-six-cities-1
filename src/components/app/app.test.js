import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import {App} from './app.jsx';


const renderer = new ShallowRenderer();

describe(`App`, () => {
  it(`renders correctly`, () => {
    renderer.render(
        <App
        />
    );
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
