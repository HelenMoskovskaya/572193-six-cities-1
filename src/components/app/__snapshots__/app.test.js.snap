// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Switch>
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(MainPage)",
          "type": [Function],
        }
      }
      exact={true}
      path="/"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(WithAuthorization)",
          "type": [Function],
        }
      }
      path="/login"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(FavoritesPage)",
          "type": [Function],
        }
      }
      path="/favorites"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(OfferPage)",
          "type": [Function],
        }
      }
      path="/offer/:id"
    />
  </Switch>
</Router>
`;
